client/src/App.jsx

```diff
@@ -54,6 +54,7 @@ function App() {
                 </Route>
                 <Route path='/network' element={<Network />} />
                 <Route path='/tester' element={<TesterComponent />} />
+                <Route path="*" element={<NotFound />} />
```

Comment:
This should be at the top of the tree for routes, that way even a mistype of a url before logging in will trigger the 404 error.

Replace line 34 with this.

Also, when I pulled your code; You are also missing the import statement for this component, please add it.